src/components/select/select.template.tsx

```diff
@@ -248,19 +248,21 @@ function createMenuItems(
     options: Option[],
     value: Option | Option[]
 ): Array<ListItem<Option>> {
-    return options.map((option) => {
-        const selected = isSelected(option, value);
-        const { text, disabled } = option;
+    return options
+        .filter((option) => !option.disabled && !!option.text)
```

Comment:
Disabled options should not be filtered out, and empty options should only be filtered out if `required` is `true`.

Reply:
The issue says

```
Expected behavior
If there is an item which has no text and it is disabled, it should not be displayed in the list of options. (If it had text, but was disabled, it could have been displayed).
```

I don't understand what you meant

Reply:
Ah! I'm sorry, I noticed a bug, and saw that you are filtering on `disabled` and `text` here, but didn't notice the `&&`.

Let me backtrack and show the bug instead of making assumptions about the underlying problem 😊

Disabled options that are _not_ empty should be visible:

Latest `next`:
<img width="827" alt="The disabled option 'Han Solo' is visible" src="https://user-images.githubusercontent.com/154725/210567568-64f5d9da-a954-41fb-990f-6654ffa2a758.png">

This PR:
<img width="827" alt="The disabled option 'Han Solo' is hidden" src="https://user-images.githubusercontent.com/154725/210567709-78bf71a8-ba54-466f-aa85-5484c2ab7a69.png">

Unless the select's @Prop `required` has been set to `true`, it should still be possible to "unset" the value, by selecting the empty option:

Latest `next`:
<img width="827" alt="Can select the empty option again after selecting an option" src="https://user-images.githubusercontent.com/154725/210568257-cdc16954-75c5-47f0-ad8d-b49f0c9a8269.png">

This PR:
<img width="827" alt="Once an option has been selected, there is no way to select 'nothing'" src="https://user-images.githubusercontent.com/154725/210568760-74c00f38-d1c2-4b39-aeb2-4f61b10b91ae.png">

Reply:
Sorry. Maybe my explanation has been bad. But Adrian summarizes the requirement  perfectly.